packages/components/src/DateTimePickers/pickers/YearPicker/YearPicker.component.js

```diff
@@ -4,61 +4,82 @@ import theme from './YearPicker.scss';
 import IconButton from '../../IconButton';
 import PickerAction from '../../PickerAction';
 
-function YearPicker(props) {
-	const years = [
-		2010,
-		2011,
-		2012,
-		2013,
-		2014,
-		2015,
-		2016,
-		2017,
-	].slice(0, 5);
+class YearPicker extends React.Component {
 
-	const selectedYear = 2012;
+	constructor(props) {
+		super(props);
 
-	function isSelected(y) {
-		return selectedYear === y;
+		const now = new Date();
+		const middleYear = props.selectedYear !== undefined
+			? props.selectedYear
+			: now.getFullYear();
+
+		const firstYear = middleYear - 2;
+
+		this.years = (new Array(5))
+			.fill(0)
+			.map((_, i) => firstYear + i);
+
+		this.onSelect = this.onSelect.bind(this);
+		this.isSelected = this.isSelected.bind(this);
 	}
 
-	return (
-		<div className={theme.container}>
-			<IconButton
-				icon={{
-					name: 'talend-chevron-left',
-					transform: 'rotate-90',
-				}}
-				className={theme['action-up']}
-				aria-label="Scroll to previous years"
-			/>
-			<div className={theme.years}>
-				{years.map((year, i) =>
-					<div
-						key={i}
-						className={theme.year}
-					>
-						<PickerAction
-							aria-label={`Select '${year}'`}
-							isSelected={isSelected(year)}
-							label={year.toString()}
-						/>
-					</div>
-				)}
+	onSelect(year) {
+		if (this.props.onSelect === undefined) {
```

Comment:
Same suggestion as for month picker, I think you can mark it as required, and drop the `if`

Reply:
same response too :p

Reply:
✔